subsys/net/lib/coap/coap.c

```diff
@@ -1095,6 +1095,10 @@ int coap_next_block_for_option(const struct coap_packet *cpkt,
 	}
 
 	coap_packet_get_payload(cpkt, &block_len);
+	/* Check that the package does not exceed the expected size */
+	if (ctx->total_size < (ctx->current + block_len)) {
```

Comment:
This isn't right, `total_size` might not be set (i. e. be 0) in case peer didn't send SIZE option.

Reply:
would this work?
```
	if (ctx->total_size > 0 && ctx->total_size < (ctx->current + block_len)) {
		return -EMSGSIZE;
	}
```
Or what do you suggest for that case?

Reply:
Yes, I think this would be fine.

Reply:
The change was is done